src/Thumbnail/FileThumbnail.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Thumbnail;
+
+use Sonata\MediaBundle\Model\MediaInterface;
+use Sonata\MediaBundle\Provider\MediaProviderInterface;
+use Symfony\Component\Asset\Packages;
+
+/**
+ * @author Jordi Sala Morales <jordism91@gmail.com>
+ */
+final class FileThumbnail implements ThumbnailInterface
+{
+    /**
+     * @var Packages
+     */
+    private $packages;
+
+    public function __construct(Packages $packages)
+    {
+        $this->packages = $packages;
+    }
+
+    public function generatePublicUrl(MediaProviderInterface $provider, MediaInterface $media, string $format): string
+    {
+        if (MediaProviderInterface::FORMAT_ADMIN !== $format) {
+            throw new \RuntimeException(sprintf('Unable to generate thumbnail for the given format %s.', $format));
```

Comment:
Shouldn't it be an `InvalidArgumentException` or something like `Unsupported format` ?

Reply:
And `InvalidArgumentException` might be the behavior of other Thumbmail instead of returning `''`, for instance in `FormatThumbnail`, when the id is null.